networkx/algorithms/tests/test_regular.py

```diff
@@ -6,46 +6,36 @@
 
 
 class TestKFactor:
-    def test_k_factor_trivial(self):
-        g = gen.cycle_graph(4)
-        f = reg.k_factor(g, 2)
-        assert g.edges == f.edges
-
-    def test_k_factor1(self):
-        g = gen.grid_2d_graph(4, 4)
-        g_kf = reg.k_factor(g, 2)
-        for edge in g_kf.edges():
-            assert g.has_edge(edge[0], edge[1])
-        for _, degree in g_kf.degree():
-            assert degree == 2
-
-    def test_k_factor2(self):
-        g = gen.complete_graph(6)
-        g_kf = reg.k_factor(g, 3)
```

Comment:
Let's make sure this (and the associated k=2 test below) are also kept.

In general I'm -1 on removing unit tests unless there's an explicit reason to do so. Unit tests are the first line of defense against backward compatibility breaks, and it's not always clear exactly why any particular test was added.